pkg/executor/insert_common.go

```diff
@@ -1256,42 +1277,44 @@ CheckAndInsert:
 			}
 		}
 
+		rowInserted := false
 		for _, uk := range r.uniqueKeys {
```

Comment:
Could we do the logic as follows? Then we can remove line1296-1317.
```
ukNewKey := uk.newKey
if tablecodec.IsTempIndexKey(ukNewKey) {
    tablecodec.TempIndexKey2IndexKey(ukNewKey)
}
```

Reply:
No, the logic changes.

Reply:
Got it. In the case of tmpIdx, we need to check both scenarios, not just one.